messages.js

```diff
@@ -0,0 +1,115 @@
+To remove gulp {{locally}} and setup gulp {{globally}}, run the following command:\n\
+  {{setupGlobalGulpCmdSubTemplate}}';
+
+/**
+ * Command that the user can run to execute gulp tasks.
+ * @type {String}
+ */
+var localGulpCmdRaw = '{{prompt}} npm run gulp <task>';
+
+/**
+ * Command that the user can run to uninstall a local gulp installation
+ * and install gulp globally.
+ * @type {String}
+ */
+var setupGlobalGulpCmdRaw = '{{prompt}} npm uninstall gulp && npm install -g gulp';
+
+/**
+ * Template for the message output from logWarningMessage.
+ * @type {String}
+ */
+var warningMessageRaw = '{{warningSign}} {{message}}';
+
+/**
+ * Object containing context for template compilation
+ * and functions for text stylingContext.
+ * @type {Object}
+ */
+var stylingContext = {
+  warningSign: color.bold.cyanBright('<' + color.bold.redBright('!') + '>'),
+  prompt: color.cyanBright('$'),
+  locally: color.yellow('locally'),
+  globally: color.yellow('globally')
+};
+
+/**
+ * Styles command strings
+ * @type {Function}
+ */
+var styleCommand = color.magenta;
+
+/**
+ * Styles warning strings
+ * @type {[type]}
+ */
+var styleWarning = color.cyanBright;
+
+/**
+ * Creates a new object containing both the stylingContext
+ * and additional elements specific to the template
+ * @param  {Object} templateContext object containing elements specific to given template
+ * @return {Object}                 the full context object to be used to build the styled template
+ */
+function generateContext(templateContext) {
+  return _.extend({}, stylingContext, templateContext);
+}
+
+/**
+ * Compiles all the templates for the install warning that is triggered
+ * when no global installation of gulp is detected
+ * @return {String} the styled message for use in yeoman installation
+ */
+function compileGulpInstallWarningMessage() {
+   var localGulpCmdTemplate = _.template(localGulpCmdRaw),
+      setupGlobalGulpCmdTemplate = _.template(setupGlobalGulpCmdRaw),
+      gulpInstallWarningTemplate = _.template(gulpInstallWarningRaw),
+      localGulpCmd, setupGlobalGulpCmd, gulpInstallWarning, context;
+
+  localGulpCmd = styleCommand(localGulpCmdTemplate(stylingContext));
+  setupGlobalGulpCmd = styleCommand(setupGlobalGulpCmdTemplate(stylingContext));
+
+  context = generateContext({
+              localGulpCmdSubTemplate: localGulpCmd,
+              setupGlobalGulpCmdSubTemplate: setupGlobalGulpCmd
+            });
+
+  return styleWarning(gulpInstallWarningTemplate(context));
+}
+
+/**
+ * Compiles the warning message template using the given message
+ * @param  {String} message to be styled and injected into the proper template
+ * @return {String}         the styled warning message
+ */
+function compileWarningMessage(message) {
+  var warningMessageTemplate = _.template(warningMessageRaw),
+      warningMessage, context;
+
+  context = generateContext({message: message});
+  return styleWarning(warningMessageTemplate(context));
```

Comment:
The things we care about in this method are
  context = ...
  warningMessage = warningMessageTemplate(context)
  return styleWarning(warningMessage);